spec/classes/unattended_upgrades_spec.rb

```diff
@@ -10,31 +10,248 @@
     :lsbrelease => '7.0.3',
   } }
 
-  it { should contain_package("unattended-upgrades") }
-
-  it { should contain_apt__conf('unattended-upgrades').with({
-      "require" => "Package[unattended-upgrades]",
-  })
-  }
-
-  it { should contain_apt__conf('periodic').with({
-      "require" => "Package[unattended-upgrades]",
-  })
-  }
-
-  it {
-    should create_file(file_unattended).with({
-      "owner"   => "root",
-      "group"   => "root",
-      "mode"    => "0644",
+  context 'with defaults' do
+    it { should contain_package('unattended-upgrades') }
+
+    it { should contain_apt__conf('unattended-upgrades').with({
+      'require' => 'Package[unattended-upgrades]',
+    })
+    }
+
+    it { should contain_apt__conf('periodic').with({
+      'require' => 'Package[unattended-upgrades]',
+    })
+    }
+
+    it {
+      should create_file(file_unattended).with({
+        'owner'   => 'root',
+        'group'   => 'root',
+        'mode'    => '0644',
+      })
+    }
+
+    it {
+      should create_file(file_periodic).with({
+        'owner'   => 'root',
+        'group'   => 'root',
+        'mode'    => '0644',
+      })
+    }
+
+    it { should contain_apt__conf('auto-upgrades').with({
+      'ensure' => 'absent',
+    })
+    }
+  end
+
+  context 'set all the things' do
+    let :params do
+      {
+        :age                  => { 'min' => 1 },
+        :size                 => { 'max' => 1000 },
+        :update               => 5,
+        :upgradeable_packages => {
+          'download_only' => 5,
+          'debdelta'      => 5,
+        },
+        :upgrade       => 5,
+        :auto          => {
+          'clean'                => '5',
+          'fix_interrupted_dpkg' => false,
+          'remove'               => false,
+          'reboot'               => false,
+        },
+        :verbose             => 1,
+        :legacy_origin       => true,
+        :origins             => [ 'bananas' ],
+        :blacklist           => [ 'foo', 'bar' ],
+        :minimal_steps       => false,
+        :install_on_shutdown => true,
+        :mail_to             => 'root@localhost',
+        :mail_only_on_error  => true,
```

Comment:
These two parameters are in the templates but have been reworked in the manifests, but it's getting too late for me now; will update in the morning.